scripts/migrate-kuzu-db.py

```diff
@@ -0,0 +1,280 @@
+    """
+    When overwrite is enabled, back up the original old_db (file with .lock and .wal or directory)
+    by renaming it to *_old, and replace it with the newly imported new_db files.
+
+    When delete_old is enabled replace the old database with the new one and delete old database
+    """
+    base_dir = os.path.dirname(old_db)
+    name = os.path.basename(old_db.rstrip(os.sep))
+    # Add _old_ and version info to backup graph database
+    backup_database_name = f"{name}_old_" + old_version.replace(".", "_")
+    backup_base = os.path.join(base_dir, backup_database_name)
+
+    if os.path.isfile(old_db):
+        # File-based database: handle main file and accompanying lock/WAL
+        for ext in ["", ".wal"]:
+            src = old_db + ext
+            dst = backup_base + ext
+            if os.path.exists(src):
+                if delete_old:
+                    os.remove(src)
+                else:
+                    os.rename(src, dst)
+                    print(f"Renamed '{src}' to '{dst}'", file=sys.stderr)
+    elif os.path.isdir(old_db):
+        # Directory-based Kuzu database
+        backup_dir = backup_base
+        if delete_old:
+            shutil.rmtree(old_db)
+        else:
+            os.rename(old_db, backup_dir)
+            print(f"Renamed directory '{old_db}' to '{backup_dir}'", file=sys.stderr)
+    else:
+        print(f"Original database path '{old_db}' not found for renaming.", file=sys.stderr)
+        sys.exit(1)
+
+    # Now move new files into place
+    for ext in ["", ".wal"]:
+        src_new = new_db + ext
+        dst_new = os.path.join(base_dir, name + ext)
+        if os.path.exists(src_new):
+            os.rename(src_new, dst_new)
+            print(f"Renamed '{src_new}' to '{dst_new}'", file=sys.stderr)
+
+
+def main():
+    p = argparse.ArgumentParser(
+        description="Migrate Kùzu DB via PyPI versions",
+        epilog="""
+Examples:
+  %(prog)s --old-version 0.9.0 --new-version 0.11.0 \\
+    --old-db /path/to/old/db --new-db /path/to/new/db --overwrite
+
+Note: This script will create temporary virtual environments in .kuzu_envs/ directory
+to isolate different Kuzu versions.
+        """,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    p.add_argument(
+        "--old-version",
+        required=False,
+        default=None,
+        help="Source Kuzu version (e.g., 0.9.0). If not provided automatic kuzu version detection will be attempted.",
+    )
+    p.add_argument("--new-version", required=True, help="Target Kuzu version (e.g., 0.11.0)")
+    p.add_argument("--old-db", required=True, help="Path to source database directory")
+    p.add_argument(
+        "--new-db",
+        required=True,
+        help="Path to target database directory, it can't be the same path as the old database. Use the overwrite flag if you want to replace the old database with the new one.",
+    )
+    p.add_argument(
```

Comment:
Thinking more about this, specifying `--overwrite` should probably automatically set `--new-db` to `old-db.tmp` or something.

But okay to keep current semantics as well.